src/NetAnalyzers/CSharp/Microsoft.CodeQuality.Analyzers/QualityGuidelines/CSharpRethrowToPreserveStackDetails.cs

```diff
@@ -62,6 +62,7 @@ private static void AnalyzeNode(SyntaxNodeAnalysisContext context)
                     case SyntaxKind.AnonymousMethodExpression:
                     case SyntaxKind.ClassDeclaration:
                     case SyntaxKind.StructDeclaration:
+                    case SyntaxKind.RecordDeclaration:
```

Comment:
Same remark as for the other PR, you will have to use the condition compilation (https://github.com/dotnet/roslyn-analyzers/pull/4223/files#diff-d8afcf96221c79be0c7c1d6e2defd475R754=

Reply:
```suggestion
#if CODEANALYSIS_V3_OR_BETTER
                    case SyntaxKind.RecordDeclaration:
#endif
```